git/repo/base.py

```diff
@@ -79,13 +104,14 @@ class Repo(object):
 
     # invariants
     # represents the configuration level of a configuration file
-    config_level = ("system", "user", "global", "repository")
+    config_level = ("system", "user", "global", "repository")  # type: Tuple[Lit_config_levels, ...]
 
     # Subclass configuration
     # Subclasses may easily bring in their own custom types by placing a constructor or type here
     GitCommandWrapperType = Git
 
-    def __init__(self, path=None, odbt=GitCmdObjectDB, search_parent_directories=False, expand_vars=True):
+    def __init__(self, path: Optional[PathLike] = None, odbt: Type[GitCmdObjectDB] = GitCmdObjectDB,
```

Comment:
LooseObjectDB?

Reply:
I think it's best to 'make up' a type and call it `ObjectDB`, as `LooseObjectDB` is too specific and kind of wrong here, too. Honestly I don't really know what I was thinking and this clearly is an example of OOP ~~being~~ gone wrong.

Otherwise you could dig into GitDB and see how this DB is composed of various types which provide a bundle of methods which may or may not be that useful in practice.

I do recommend to not get into types of GitDB and instead make them up (inside of GitPython) based on what's common usage in GitPython instead.